pkg/engine/task/task_apply.go

```diff
@@ -58,8 +60,8 @@ func (at ApplyTask) Run(ctx Context) (bool, error) {
 	// 4. - Check health for all resources -
 	err = isHealthy(applied)
 	if err != nil {
-		if fatal := isTerminallyFailed(applied); fatal != nil {
-			return false, fatalExecutionError(fatal, failedTerminalState, ctx.Meta)
+		if errors.Is(err, engine.ErrFatalExecution) {
```

Comment:
We return a `type engine.ExecutionError struct` if we encountered a fatal health error.